pkg/controllers/proxy/network_services_controller.go

```diff
@@ -812,7 +848,7 @@ func (nsc *NetworkServicesController) OnEndpointsUpdate(ep *api.Endpoints) {
 	newServiceMap := nsc.buildServicesInfo()
 	newEndpointsMap := nsc.buildEndpointsInfo()
 
-	if len(newEndpointsMap) != len(nsc.endpointsMap) || !reflect.DeepEqual(newEndpointsMap, nsc.endpointsMap) {
+	if !compatible(newEndpointsMap, nsc.endpointsMap) {
```

Comment:
Just checking here, you don't need to do something similar for the DeepEqual in `OnServiceUpdate` below?

Reply:
The [`buildServicesInfo()` method](https://github.com/cloudnativelabs/kube-router/blob/master/pkg/controllers/proxy/network_services_controller.go#L1308-L1386) doesn't shuffle the result like [`buildEndpointsInfo()`](https://github.com/cloudnativelabs/kube-router/blob/master/pkg/controllers/proxy/network_services_controller.go#L1419-L1437) does, so the DeepEqual there should hopefully do better in that comparison

Reply:
I think that you are likely right. After the two corrections above are made, I'll test it out and double-check in one of my clusters.